ayase/view/asagi.py

```diff
@@ -61,79 +76,112 @@ def custom_openapi(openapi_prefix: str):
 thumb_uri = CONF["image_location"]["thumb"]
 DB_ENGINE = CONF["database"]["default"]
 
-# This has to be here so the variables above are loaded prior to importing these
-from model.asagi import convert_thread, generate_index, convert_post, generate_gallery
-
-
 if boards:
     for i in boards:
         board_list.append(i["shortname"])
 
-for i in archives:
-    archive_list.append(i["shortname"])
+if archives:
+    for i in archives:
+        archive_list.append(i["shortname"])
+
+
+# Caches templates
+template_index = env.get_template("index.html")
+template_board_index = env.get_template("board_index.html")
+template_404 = env.get_template("404.html")
+template_gallery = env.get_template("gallery.html")
+template_thread = env.get_template("thread.html")
+template_post = env.get_template("post.html")
+template_posts = env.get_template("posts.html")
+
+
+# This has to be here so the variables above are loaded prior to importing these
+from model.asagi import (
+    convert_thread,
+    generate_index,
+    convert_post,
+    generate_gallery,
+)
 
 
 class NotFoundException(Exception):
-    def __init__(self):
-        pass
+    def __init__(self, board_name=CONF["site_name"]):
+        self.title_window = board_name
+        if board_name == CONF["site_name"]:
+            return
```

Comment:
- Fix 404 board title not defaulting to site name  

This should be replaced with
```python
        if self.title_window == CONF["site_name"]:
            return
        if not (board_name in archive_list or board_name in board_list):
            self.title_window = CONF["site_name"]
            return
```